segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/BaseChunkStorage.java

```diff
@@ -142,6 +147,26 @@ public BaseChunkStorage(Executor executor) {
      */
     abstract protected ChunkHandle doCreate(String chunkName) throws ChunkStorageException;
 
+    /**
+     * Creates a new chunk.
+     *
+     * @param chunkName String name of the chunk to create.
+     * @param length Number of bytes to write.
+     * @param data   An InputStream representing the data to write.* @return ChunkHandle A writable handle for the recently created chunk.
+     * @throws ChunkStorageException    Throws ChunkStorageException in case of I/O related exceptions.
+     * @throws IllegalArgumentException If argument is invalid.
+     */
+    protected ChunkHandle doCreateWithContent(String chunkName, int length, InputStream data) throws ChunkStorageException {
+        ChunkHandle handle = doCreate(chunkName);
+        int bytesWritten = doWrite(handle, 0, length, data);
+        if (bytesWritten < length) {
```

Comment:
What if some other error occurred? Is there any situation where we could leave garbage behind?

Also why would we write fewer bytes?